src/test/forms/rentFields/ContractRentField.test.js

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { defaultsDeep } from 'lodash';
+
+import { CLIENT_DEFAULTS } from '../../../utils/CLIENT_DEFAULTS';
+import { ContractRentField } from '../../../forms/rentFields';
+
+const buildField = (values = {}) => {
+  const props = {
+    timeState:         defaultsDeep({}, values, CLIENT_DEFAULTS.current),
+    updateClientValue: jest.fn(),
+  };
+  return mount(<ContractRentField { ...props } />);
+};
+
+describe('ContractRentField.storeValidator()', () => {
+  it('when input valid, sets state to valid', () => {
+    const field = buildField({ rentShare: 0 });
+    expect(field.state()).toEqual({ valid: true, message: null });
+
+    field.instance().storeValidator(1);
+    field.update();
+    expect(field.state()).toEqual({ valid: true, message: null });
+  });
+
+  it('when input negative, sets state to invalid', () => {
+    const field = buildField();
+    expect(field.state()).toEqual({ valid: true, message: null });
+
+    field.instance().storeValidator(-1);
+    field.update();
+    expect(field.state()).toEqual({ valid: false, message: null });
+  });
+
+  it('when input less than rent share, sets state to invalid', () => {
```

Comment:
Would it be accurate to describe this as 'when contract rent less than rent share'?

Reply:
I think so. It might be slightly different in that you have the contract rent in the client as well as the potential contract rent in the input. But just saying contract rent is probably clearer.